src/build/parse/generate.py

```diff
@@ -10,6 +10,52 @@
 from pathlib import Path
 
 
+def generate_top_level_text(interface, citations):
+
+    DEFAULT_CITATIONS = {
+        "Bilayers": [
+            {
+                "name" : "Bilayers",
+                "license" : "BSD 3-Clause",
+                "description" : "A Container Specification and CI/CD  built for whole-community support"
+            },
+        ],
+        "Jupyter": [
+            {
+                "name" : "Jupyter",
+                "doi" : "10.1109/MCSE.2007.53",
+                "license" : "BSD 3-Clause",
+                "description" : "Interactive, code-driven documents for data analysis and visualization"
+            },
+        ],
+        "Gradio": [
+            {
+                "name" : "Gradio",
+                "doi" : "10.48550/arXiv.190602569",
+                "license" : "Apache License 2.0",
+                "description" : "A simple web interface for deploying machine learning models"
+            },
+        ],
+          }
+    app_descriptions = "This interface provides the following tool(s):\n"
+    citation_text = "This project relies on citations! Please cite ALL of the following if you find this application useful in your research!\n"
+    license_info = "The licenses of the components of this project are provided below. Please ensure these licenses are compatible with your work before using these tools.\n"
+    app_names = ""
+    for citation in citations:
+        app_descriptions+= f"{ citation["name"] }: { citation["description"] }\n" 
+        citation_text += f"Cite { citation["name"] } using { citation["doi"] }\n"
+        license_info += f"{ citation["name"] } is provided under the { citation["license"] } license\n"
+        app_names += f"{ citation["name"] }+"
+    for default_thing in ["Bilayers",interface]:
+        citation = DEFAULT_CITATIONS[default_thing]
+        if "doi" in citation.keys():
```

Comment:
can't call `keys()` on a list

Reply:
Miiiiiinor detail. Do you recall/can you think of a good reason of why this would be a list? Do we want to support multiple-entries for the citations (or parts of the citations, like DOIs?)

Reply:
I don't recall. My guess is for the tool itself, and any algorithms it uses (e.g. a DL tool, that itself uses OpenCV for post processing, and further uses a citable algorithm in OpenCV like SIFT)? 

Like
```
"algorithm": [
    {
        "name": "my_cool_dl_thing",
        ...
    },
    {
        "name": "some_lib_used_in_postprocessing",
        ....
    },
    {
        "name": "some_algo_used_by_lib_in_postprocessing",
        ....
    }
] 
```

Reply:
Which happens not to hold for `DEFAULT_CITATIONS` but may hold for citations gathered from a `config.yaml`

Reply:
Yeahhhhh so I should make a list in both places. 10/4.